controllers/namespacelabel_controller.go

```diff
@@ -66,81 +64,101 @@ type NamespaceLabelReconciler struct {
 // This reconcile function adds the labels from the NamespaceLabel to the namespace it runs against,
 // and deletes the labels when the resource is deleted.
 func (r *NamespaceLabelReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := logrus.New()
-	log.SetFormatter(&ecslogrus.Formatter{})
-
+	log := ctrllog.FromContext(ctx)
 	var namespaceLabel danaiodanaiov1alpha1.NamespaceLabel
 	if err := r.Get(ctx, req.NamespacedName, &namespaceLabel); err != nil {
 		if k8serrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.WithError(errors.New(err.Error())).WithFields(logrus.Fields{"NamespaceLabel": req.NamespacedName}).Error("unable to fetch NamespaceLabel")
+		log.Error(err, "Unable to fetch NamespaceLabel", "namespaceLabel", req.NamespacedName.String())
 		return ctrl.Result{Requeue: true}, client.IgnoreNotFound(err)
 	}
 
 	var namespace corev1.Namespace
 	if err := r.Client.Get(ctx, types.NamespacedName{Name: namespaceLabel.ObjectMeta.Namespace}, &namespace); err != nil {
-		log.WithError(errors.New(err.Error())).WithFields(logrus.Fields{"namespace": namespaceLabel.ObjectMeta.Namespace}).Error("unable to fetch namespace while getting previous labels")
+		log.Error(err, "Unable to fetch namespace", "namespace", namespaceLabel.ObjectMeta.Namespace)
 		return ctrl.Result{Requeue: true}, client.IgnoreNotFound(err)
 	}
+	if namespaceLabel.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&namespaceLabel, FinalizerName) {
+			if err := r.addFinalizer(ctx, log, &namespaceLabel, &namespace); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if controllerutil.ContainsFinalizer(&namespaceLabel, FinalizerName) {
+			if res, err := r.deleteLabels(ctx, log, &namespaceLabel, &namespace); err != nil {
+				return res, err
+			}
+			return ctrl.Result{}, nil
+		}
+	}
+	if err := r.updateLabels(ctx, log, &namespaceLabel, &namespace); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
 
+func (r *NamespaceLabelReconciler) updateLabels(ctx context.Context, log logr.Logger, namespaceLabel *danaiodanaiov1alpha1.NamespaceLabel, namespace *corev1.Namespace) error {
 	labels := namespace.GetLabels()
-	protectedLabels := getProtectedLabels()
+	protectedPrefixes := getProtectedPrefixes(r.ProtectedPrefixes)
 	for key, val := range namespaceLabel.Spec.Labels {
-		if !utils.Contains(protectedLabels, key) {
+		_, exists := labels[key]
+		if !utils.Contains(protectedPrefixes, strings.Split(key, "/")[0]) && !exists {
 			labels[key] = val
+			condition := metav1.Condition{Type: "LabelApplied", Status: "True", Reason: "LabelApplied", Message: "Label " + key + " = " + val + " was applied", LastTransitionTime: metav1.Time{Time: time.Now()}}
+			namespaceLabel.Status.Conditions = append(namespaceLabel.Status.Conditions, condition)
```

Comment:
The recommended way to update a condition, as explained in the operator-sdk website, is through SetStatusCondition ([code](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L30-L55)) and not updating the entire object:

Reply:
updated in latest pr

Reply:
You don't necessarily need a new PR for that, you could have just made a new commit (or squash the commits and force push the latest change to this branch).